browser/net/brave_network_delegate_base.cc

```diff
@@ -102,6 +121,30 @@ void BraveNetworkDelegateBase::InitPrefChangeRegistrarOnUI() {
                  base::Unretained(this)));
   // Retrieve current referral headers, if any.
   OnReferralHeadersChanged();
+
+  PrefService* user_prefs = ProfileManager::GetActiveUserProfile()->GetPrefs();
+  user_pref_change_registrar_.reset(new PrefChangeRegistrar());
+  user_pref_change_registrar_->Init(user_prefs);
+  user_pref_change_registrar_->Add(
+      kGoogleLoginControlType,
+      base::Bind(&BraveNetworkDelegateBase::OnPreferenceChanged,
+                 base::Unretained(this), kGoogleLoginControlType));
+  user_pref_change_registrar_->Add(
+      kFBEmbedControlType,
+      base::Bind(&BraveNetworkDelegateBase::OnPreferenceChanged,
+                 base::Unretained(this), kFBEmbedControlType));
+  user_pref_change_registrar_->Add(
+      kTwitterEmbedControlType,
+      base::Bind(&BraveNetworkDelegateBase::OnPreferenceChanged,
+                 base::Unretained(this), kTwitterEmbedControlType));
+  user_pref_change_registrar_->Add(
```

Comment:
I first tried adding these observers inside of the ad block base class but it caused problems because of threading, it doesn't work from the main thread which prefs needed to.